shm/src/Ship.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Ship.hpp"
+
+Ship::Ship()
+    : id_(-1) {}
+Ship::Ship(uint16_t id, const std::string& name, uint16_t speed, uint16_t maxCrew, uint16_t capacity)
+    : id_(id), name_(name), speed_(speed), maxCrew_(maxCrew), capacity_(capacity), crew_(0) {}
+Ship::Ship(uint16_t id, uint16_t speed, uint16_t maxCrew)
+    : Ship(id, "", speed, maxCrew, 0) {}
+
+void Ship::setName(const std::string& name) {
+    name_ = name;
+}
+
+Ship& Ship::operator+=(uint16_t newCrewMembers) {
+    if (crew_ + newCrewMembers > maxCrew_) {
```

Comment:
I think it's better to compare crew_ and newCrewMembers, because if we add crew_ and newCrewMembers we risk going out of uint16_t range,which may result in gargabe value that can be lower than crew limit.